rcl/src/rcl/logging_rosout.c

```diff
@@ -153,8 +154,11 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(
     return RCL_RET_ERROR;
   }
   if (rcutils_hash_map_key_exists(&__logger_map, &logger_name)) {
-    RCL_SET_ERROR_MSG("Logger already initialized for node.");
-    return RCL_RET_ALREADY_INIT;
+    // @todo: nburek Update behavior to either enforce unique names or work with non-unique 
+    // names based on the outcome here: https://github.com/ros2/design/issues/187
+    RCUTILS_LOG_WARN_NAMED("log_infrastructure", "Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from going out over the rosout topic.");
+    // RCL_SET_ERROR_MSG("Logger already initialized for node.");
+    // return RCL_RET_ALREADY_INIT;
```

Comment:
I think we want to return here to avoid a possible memory leak.
Otherwise, we replace an reference to an existing `rcl_publisher_t` without finalizing it.

```suggestion
    return RCL_RET_OK;
```

Reply:
I think this will also have the desired behavior that the first node created for a given name will be the one associated with the publisher.